File: solution/1208. Get Equal Substrings Within Budget/1208.cpp
class Solution {
public:
    int equalSubstring(string s, string t, int maxCost) {
        const int n = s.size();
        int res = 0;
        for (int j = 0, i = 0; i < n; ++i) {
            int diff = abs(s[i] - t[i]);
            while (diff > maxCost) {
                maxCost += abs(s[j] - t[j]);
                ++j;
            }
            if (diff <= maxCost) {
                res = max(res, i - j + 1);
                maxCost -= diff;
            }
        }
        return res;
    }
};

// a b c d
// b c d f
// 1 1 1 2

File: solution/1208. Get Equal Substrings Within Budget/1208.py
class Solution:
    def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
        j = 0
        for i in range(len(s)):
            maxCost -= abs(ord(s[i]) - ord(t[i]))
            if maxCost < 0:
                maxCost += abs(ord(s[j]) - ord(t[j]))
                j += 1
        return i - j + 1 # python 的 for 循環只會去到 n-1
        
